启动游戏参数

baimaoshouerkong opened the pull request:
这个差不多有个解决启动参数的方案
<img width="572" alt="image" src="https://github.com/user-attachments/assets/7300f77c-f8f9-4bb3-a17e-71149ced512e">
和cmd方案差不多
具体的实现方式就是在galgame类中加入Startup_parameters的参数
在galgameviewmodel中Play方法中判断是否Startup_parameters为空字符，

然后使用正则表达式".+?"，匹配地址，剩下的字符串作为参数

actions-user committed:
Update data

baimaoshouerkong committed:
Merge branch 'GoldenPotato137:dev' into dev

actions-user committed:
Update data

actions-user committed:
Update data

baimaoshouerkong committed:
Merge branch 'GoldenPotato137:dev' into dev

baimaoshouerkong committed:
先搞个分支

第一步，新建文件夹（汗）

baimaoshouerkong committed:
现在大概支持启动参数启动了吧(汗)

我为galgame中加入一个名为_startup_parameters的变量，在游戏设置里我加入启动参数的文本框，两者进行绑定，如果_startup_parameters为空字符那就按原方法启动，如果不是那就启动cmd并将参数写入。（等一下测试）

actions-user committed:
Update data

baimaoshouerkong committed:
修改了一下逻辑

突然发现使用cmd后出现了各种各样的问题。
感觉还是用软件去打开以其他软件的参数去打开程序，问题会少一点

Commit:
Merge branch 'GoldenPotato137:dev' into dev

baimaoshouerkong committed:
错误修改

代码有点问题（汗）

baimaoshouerkong committed:
emmm，修改错误

难崩

baimaoshouerkong committed:
Merge branch 'dev' into 启动游戏参数

GoldenPotato137 commented on GalgameManager/ViewModels/GalgameViewModel.cs:
```diff
@@ -217,22 +218,44 @@ private async Task OpenInYmgal()
     {
-        if (Item == null) return; //不应该发生
-        if (Item.ExePath == null)
+        if (Item is not {SourceType:GalgameSourceType.LocalFolder}) return;
```
不要用Galgame.SourceType，它是个废弃的字段（原本计划这段时间处理来着）。
以前原来的游戏是只能同时属于一个源的（比如说如果一个游戏是本地文件夹游戏，那么它不能同时属于本地压缩库（还没做完）、各种云盘库（还没做）等）。
新的库机制允许一个游戏同时属于多个库了（可以看看Galgame.Sources）。因此，要判断一个游戏是不是本地文件夹游戏，应该用Galgame.IsLocalGame

以及目前Galgame.Path也是个应该废弃的字段（原因和上面一样），应该从Galgame.Sources中找到本地游戏库（每个游戏只能属于某一个本地游戏库），再通过这个source获取这个游戏在source中的路径：比如说从本地游戏库获取游戏的路径就是原来的Path。

p.s. Path废弃造成的超级屎山还在处理，你写的新功能中涉及Path的地方可以帮忙修改修改

baimaoshouerkong replied:
这个合并冲突的时，直接合并了，没有看，我尽力

GoldenPotato137 commented on GalgameManager/ViewModels/GalgameViewModel.cs:
```diff
@@ -217,22 +218,44 @@ private async Task OpenInYmgal()
-        if (Item == null) return; //不应该发生
-        if (Item.ExePath == null)
+        if (Item is not {SourceType:GalgameSourceType.LocalFolder}) return;
+        if (Item.ExePath == null && Item.Startup_parameters=="")
```
感觉可以用string.empty代替""，看起来美观些（

baimaoshouerkong replied:
OK，我马上改

GoldenPotato137 commented on GalgameManager/ViewModels/GalgameViewModel.cs:
```diff
@@ -217,22 +218,44 @@ private async Task OpenInYmgal()
 
-        Item.LastPlayTime = DateTime.Now;
-        Process process = new()
+        Item.LastPlay = DateTime.Now.ToShortDateString();
```
LastPlay已经废弃啦，看看原来的写法。
![image](https://github.com/user-attachments/assets/ec9ac334-27de-4481-9050-458d21735298)

GoldenPotato137 commented on GalgameManager/ViewModels/GalgameViewModel.cs:
```diff
@@ -217,22 +218,44 @@ private async Task OpenInYmgal()
+                {
+                    FileName = filename,
+                    Arguments = arguments,
+                    UseShellExecute = false,
```
这里是不能用shell来执行嘛？如果需要管理员启动游戏该咋处理捏？

baimaoshouerkong replied:
有道理，我改改

GoldenPotato137 commented on GalgameManager/Views/GalgameSettingPage.xaml:
```diff
@@ -89,6 +89,24 @@
+                            <ColumnDefinition Width="Auto" />
+                            <ColumnDefinition Width="*" />
+                        </Grid.ColumnDefinitions>
+                        <TextBlock Text="启动参数设置"
```
这里最好还是用resource处理一下，要不到时候想本地化的时候又是屎山

baimaoshouerkong replied:
好的

baimaoshouerkong replied:
OK

baimaoshouerkong committed:
修改一点逻辑

启动游戏后因为启动进程和游戏进程不是同一个进程，所以在计算时间的时候可能会出现问题，还有一点小小的问题，剩下的问题明天搞

baimaoshouerkong committed:
Update GalgameViewModel.cs

baimaoshouerkong committed:
正则表达式规则更新，The value cannot be an empty string (Pararmeter'oldValue')bug修复

更新了正则表达式，支持环境变量的使用，修复了一下逻辑，把The value cannot be an empty string (Pararmeter'oldValue')bug修复

baimaoshouerkong commented:
> 这个差不多有个解决启动参数的方案 <img alt="image" width="572" src="https://private-user-images.githubusercontent.com/127100434/376871327-7300f77c-f8f9-4bb3-a17e-71149ced512e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyMTU1MTgsIm5iZiI6MTcyOTIxNTIxOCwicGF0aCI6Ii8xMjcxMDA0MzQvMzc2ODcxMzI3LTczMDBmNzdjLWY4ZjktNGJiMy1hMTdlLTcxMTQ5Y2VkNTEyZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxOFQwMTMzMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZmU3OTYwNTQ2YzNiODI2ZTY0MDJjZWY3NDlkMjg3YzU1YTRlMjIwNGZhNjRlMzFkY2QxOTkyZGVhN2VkYzRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.334ld32ibULoD9C8d_-MKKeKQJLHc4BBachtX_hKAJw"> 和cmd方案差不多 具体的实现方式就是在galgame类中加入Startup_parameters的参数 在galgameviewmodel中Play方法中判断是否Startup_parameters为空字符，
> 
> 然后使用正则表达式".+?"，匹配地址，剩下的字符串作为参数
修改了一下逻辑，系统的环境变量的方式支持

GoldenPotato137 commented:
图片挂啦，看不到

baimaoshouerkong commented:
> 图片挂啦，看不到

好了，我周末玩兰斯去了，没有看github

baimaoshouerkong committed:
Update ActivationService.cs

Commit:
Merge branch 'dev' into 启动游戏参数

baimaoshouerkong committed:
处理冲突

baimaoshouerkong commented:
> 图片挂啦，看不到

ok修好了

Commit:
Merge branch 'dev' into 启动游戏参数

GoldenPotato137 commented on GalgameManager/Services/GalgameCollectionService/GalgameCollectionService.cs:
```diff
@@ -615,6 +615,21 @@ private async Task OnSettingChanged(string key)
+        _ = _bgTaskService.AddBgTask(new UnpackGameTask(file, folderSource, dialog.GameName, dialog.Password));
+    }
+
+
```
merge时出锅啦，这里是要删掉哒

baimaoshouerkong replied:
ok

GoldenPotato137 replied:
hxd，删的时候看看git记录呀，这里是要把整个ToLocalGalgame删掉 （

baimaoshouerkong replied:
我的问题，我看网页上就是这行代码

GoldenPotato137 commented on GalgameManager/Views/GalgamePage.xaml:
```diff
@@ -73,6 +73,18 @@
+                        <FontIcon FontFamily="Segoe Fluent Icons" Glyph="&#xF012;" />
+                    </AppBarButton.Icon>
+                </AppBarButton>
+
```
merge出锅啦，这里是要删掉的内容

baimaoshouerkong replied:
ok

GoldenPotato137 replied:
这里也是一样的，要把整个解压压缩游戏的AppBarButton去掉

baimaoshouerkong replied:
我是傻逼

GoldenPotato137 reviewed (commented):
目测没啥问题啦，把那几个merge出锅的地方处理一下后我就merge进来一版，感谢贡献~

baimaoshouerkong committed:
把冲突删去了

baimaoshouerkong committed:
Merge branch '启动游戏参数' of https://github.com/baimaoshouerkong/PotatoVN into 启动游戏参数

baimaoshouerkong committed:
emmm

GoldenPotato137 reviewed (approved):
没问题啦，感谢贡献~
